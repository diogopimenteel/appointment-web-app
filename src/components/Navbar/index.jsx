import React from 'react';
import { Navbar, Image } from 'react-bootstrap';

function index() {
  return (
    <Navbar sticky="top" bg="dark" variant="dark" expand="lg" data-testid="navbar-test">
      <Navbar.Brand href="/">
        <Image
          className="d-inline-block align-center"
          src="/images/logo.jpg"
          thumbnail="true"
          roundedCircle="true"
          width="65"
          height="65"
        />
        {' '}
        Appointments for COVID-19 vaccination

      </Navbar.Brand>
    </Navbar>
  );
}

export default index;
