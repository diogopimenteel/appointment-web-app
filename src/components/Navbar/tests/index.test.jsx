/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import ReactDOM from 'react-dom';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';
import Navbar from '../index';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navbar />, div);
});

it('renders navbar correctly', () => {
  const { getByTestId } = render(<Navbar />);
  expect(getByTestId('navbar-test')).toHaveTextContent('Appointments for COVID-19 vaccination');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
