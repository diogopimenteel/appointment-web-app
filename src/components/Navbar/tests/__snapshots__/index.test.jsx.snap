// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top"
  data-testid="navbar-test"
>
  <a
    className="navbar-brand"
    href="/"
  >
    <img
      className="d-inline-block align-center rounded-circle img-thumbnail"
      height="65"
      src="/images/logo.jpg"
      width="65"
    />
     
    Appointments for COVID-19 vaccination
  </a>
</nav>
`;
